### PR TITLE
Refactor some utilities and TODO done

### PR DESCRIPTION
**Pull request by Ariel-Isaacm**
Hey guys, awesome framework +1 i've been looking at the code and i did some improvements, also i did a TODO but i'm not quite sure how to handle that null check it would be awesome if you could check it out @bric3 @szczepiq 

Answers.java : private is not required 
ClassPathLoader.java: Removed unnecessary imports
ScenarioPrinter.java: Changed the string concat into another append
MockHandlerFactory.java: Return statement changed to be inline
ValuePrinter.java: TODO done, null check added

Edit 
ReleaseNotesExtension : Made authToken a final String

Edit 2: 
Some improvements to the ThreadTests :)

### CODE REVIEW

File: src/test/java/org/concurrentmockito/ThreadsRunAllTestsHalfManualTest.java [hunk 1]
```diff
@@ -170,7 +170,7 @@ public static boolean runInMultipleThreads(int numberOfThreads) throws Exception
         boolean failed = false;
         for (AllTestsRunner t : threads) {
             t.join();
-            failed = failed ? true : t.isFailed();
+            failed = failed || t.isFailed();
```
Comment (JeremybellEU) [inline src/test/java/org/concurrentmockito/ThreadsRunAllTestsHalfManualTest.java]: Does a `|=` also work?

File: src/main/java/org/mockito/internal/matchers/text/ValuePrinter.java [hunk 2]
```diff
@@ -41,7 +41,9 @@ public static String print(Object value) {
      * @return neatly formatted value list
      */
     public static String printValues(String start, String separator, String end, Iterator values) {
-        //TODO SF null check
+        if(start == null || separator == null || end == null || values == null){
+            return null;
```
Comment (JeremybellEU) [inline src/main/java/org/mockito/internal/matchers/text/ValuePrinter.java]: Returning null seems dangerous for future usages. Maybe it is better to return empty string or use an exception to notigy the developer he made an error.

Comment (Ariel-Isaacm) [inline src/main/java/org/mockito/internal/matchers/text/ValuePrinter.java]: You're right :o! i'll put a empty string for now, thanks!

File: src/test/java/org/concurrentmockito/ThreadsRunAllTestsHalfManualTest.java (hunk 1 above)
Comment (Ariel-Isaacm) [inline src/test/java/org/concurrentmockito/ThreadsRunAllTestsHalfManualTest.java]: Yes! you're right i already made the changes thanks for your comments :)

**Comment by Ariel-Isaacm** [comment]
Commits squashed*

**Comment by pmduaree** [comment]
Good job! The refactor is well done and the code is cleaner now. 

Maybe the coverage is gonna be decreased thanks to the null check. Maybe a new test will do.

File: src/main/java/org/mockito/internal/matchers/text/ValuePrinter.java [hunk 3]
```diff
@@ -41,7 +41,10 @@ public static String print(Object value) {
      * @return neatly formatted value list
      */
     public static String printValues(String start, String separator, String end, Iterator values) {
-        //TODO SF null check
+        if(start == null || separator == null || end == null || values == null){
+            return "";
+            //TODO handle this situation
+        }
```
Comment (bric3) [inline src/main/java/org/mockito/internal/matchers/text/ValuePrinter.java]: I think you can just replace `null` by some default values of each fields

**Comment by bric3** [comment]
Looks good to me, I think you can just fixup the todo with my [comment](https://github.com/mockito/mockito/pull/301#discussion_r46943080)

**Comment by Ariel-Isaacm** [comment]
Hi, i did the changes, are they what you expected? and should i squash the two commits?

**Comment by raphw** [comment]
Yes please. Thank you!

**Comment by Ariel-Isaacm** [comment]
Done :)

**Comment by bric3** [comment]
Thanks @Ariel-Isaacm for the contribution

**Comment by Ariel-Isaacm** [comment]
Thanks you for your patience :) @bric3 @raphw

**Comment by raphw** [comment]
No, thank you!

**Comment by bric3** [comment]
Yes exactly it was quite the opposite, thanks for the patience ;)

**Pull request merged**